docs/source/package_reference/hotswap.md

```diff
@@ -0,0 +1,45 @@
+<!--⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+-->
+
+# Hotswapping adapters
+
+The idea of hotswapping an adapter is the following: We can already load multiple adapters, e.g. two LoRAs, at the same time. But sometimes, we want to load one LoRA and then replace its weights in-place with the LoRA weights of another adapter. This is now possible the `hotswap_adapter` function.
+
+In general, this should be faster than deleting one adapter and loading the adapter in its place, which would be the how to achieve the same final outcome without hotswapping. Another advantage of hotswapping is that it prevents re-compilation in case the PEFT model is already compiled using `torch.compile`. This can save quite a lot of time.
+
+```python
+import torch
+from transformers import AutoModelForCausalLM
+from peft import PeftModel
+from peft.utils.hotswap import hotswap_adapter
+
+model_id = ...
+inputs = ...
+device = ...
+model = AutoModelForCausalLM.from_pretrained(model_id).to(device)
+
+# load lora 0
+model = PeftModel.from_pretrained(model, <path-adapter-0>)
+model = torch.compile(model)  # optionally compile the model
+with torch.inference_mode():
+    output_adapter_0 = model(inputs)
+
+# replace the "default" lora adapter with the new one
+hotswap_adapter(model, <path-adapter-1>, adapter_name="default", torch_device=device)
+with torch.inference_mode():
+    output_adapter_1 = model(inputs).logits
+```
+
+
+There are some caveats for hotswapping:
+
+- It only works for the same PEFT method, so no swapping LoRA and LoHa, for example.
+- Right now, only LoRA is properly supported.
+- The adapters must be compatible (e.g. same LoRA alpha, same target modules).
+
```

Comment:
Could add a note saying this is not limited to `transformers` and works with `diffusers`, too. But if we wanna wait until https://github.com/huggingface/diffusers/pull/9453 is figured out and merged, I will understand.

Reply:
I added a sentence. It should already work with diffusers models when users use the `hotswap_adapter` function, it's just not natively in diffusers yet, so I'm fine with adding it.